### ISSUE TITLE
Any interest in adding more thorough automated tests?

### ISSUE THREAD

**Issue by sgrif** [macroid/macroid#52]
I'd be happy to do the work on this, but I figured I'd gauge interest before I take the leap. I think this project could benefit greatly from a set of instrumentation tests, running through the UI, verified via Espresso. It'd allow much more thorough testing of each public method individually (and presumably we could get a Travis CI setup going as well). Would there be any interest in a series of pull requests to add this?

**Comment by stanch** [comment]
Yes, definitely! *Macroid* was born as I factored out some pieces of a relatively big application, and I ended up developing both of them very closely and effectively using that app as a test suite. That’s really a shame that the library now has only “compilation” tests and I’d be happy to see it tested in more creative ways :)

**Comment by sgrif** [comment]
Cool. Rails takes up most of my open source free time but I'll try to start putting some time in towards this.
